### PR TITLE
Enable network gremlin

### PR DESCRIPTION
**Pull request by dpritchett**
Thanks @philgebhardt and the Pivotal team for helping me chase this one down.

### CODE REVIEW

File: jobs/gremlind/templates/pre-start.erb [hunk 1]
```diff
@@ -40,8 +40,9 @@ place-binaries() {
 
     # empower gremlin client to perform OS functions: see https://help.gremlin.com/security/#linux-capabilities
     setcap cap_sys_boot,cap_kill,cap_sys_time,cap_net_admin+ep /usr/bin/gremlin
+    sudo setcap cap_sys_boot,cap_kill,cap_sys_time,cap_net_admin+ep /usr/bin/gremlin
```
Comment (philgebhardt) [inline jobs/gremlind/templates/pre-start.erb]: duplicate line from above (with `sudo`, which I don't think we need)

File: jobs/gremlind/templates/pre-start.erb [hunk 2]
```diff
@@ -40,8 +40,9 @@ place-binaries() {
...
     # enable networking attacks
-    setcap cap_net_admin+ep /sbin/tc
+    sudo setcap cap_net_admin+ei $(which tc)
```
Comment (philgebhardt) [inline jobs/gremlind/templates/pre-start.erb]: probably don't need sudo

File: jobs/gremlind/templates/pre-start.erb [hunk 3]
```diff
@@ -61,14 +62,32 @@ create-user() {
     # create only if not already created
     id -u gremlin &>/dev/null || useradd --system --user-group --home /var/lib/gremlin gremlin
 
-    groups gremlin | grep &>/dev/null "gremlin$" || usermod -aG gremlin
+    groups gremlin | grep "gremlin.*gremlin" || usermod -a -G gremlin gremlin
+    groups gremlin | grep "vcap" || usermod -a -G vcap gremlin
```
Comment (philgebhardt) [inline jobs/gremlind/templates/pre-start.erb]: I don't think we need `vcap` added to `gremlin`

File: jobs/gremlind/templates/pre-start.erb [hunk 4]
```diff
@@ -61,14 +62,32 @@ create-user() {
...
+
+	# Create the gremlin system user
```
Comment (philgebhardt) [inline jobs/gremlind/templates/pre-start.erb]: duplicate code? we can remove this

File: jobs/gremlind/templates/pre-start.erb [hunk 5]
```diff
@@ -61,14 +62,32 @@ create-user() {
...
+	#sudo useradd --system --user-group --home /var/lib/gremlin gremlin
 }
 
 prepare-file-system() {
     mkdir -p /var/lib/gremlin
     mkdir -p /var/log/gremlin
     chown -R gremlin:gremlin /var/lib/gremlin
     chown -R gremlin:gremlin /var/log/gremlin
+
+    # Create the needed directories
```
Comment (philgebhardt) [inline jobs/gremlind/templates/pre-start.erb]: same comment RE: `sudo`

**Review by philgebhardt — COMMENTED** [review]
Please remove the calls to `sudo` and the duplicate line of code

File: VERSION [hunk 6]
```diff
@@ -1 +1 @@
-0.1.15
+0.1.18
```
Comment (philgebhardt) [inline VERSION]: why are we bumping `15 -> 18`?

Comment (dpritchett) [reply VERSION]: The intermediate versions are in earlier commits to this branch - just need a discrete version number for each new attempt on PIE.

**Review by philgebhardt — APPROVED** [review]
Looks good!

File: jobs/gremlind/templates/pre-start.erb [hunk 7]
```diff
@@ -43,6 +43,7 @@ place-binaries() {
 
     # enable networking attacks
     setcap cap_net_admin+ei $(which tc)
+    setcap cap_net_admin+ei /bin/ip # /sbin/ip is a symlink to this
```
Comment (philgebhardt) [inline jobs/gremlind/templates/pre-start.erb]: I don't think we need this

**Review by philgebhardt — CHANGES_REQUESTED** [review]
See comment about removing caps from `ip`

**Review by philgebhardt — APPROVED** [review]

**Pull request merged**